tensorflow/contrib/tensorrt/kernels/trt_engine_op.cc

```diff
@@ -236,32 +244,30 @@ void TRTEngineOp::ExecuteCalibration(OpKernelContext* ctx,
   ExecuteNativeSegment(ctx, helper);
 }
 
-int TRTEngineOp::GetEngineBatch(OpKernelContext* ctx) {
-  int num_batch = ctx->input(0).shape().dim_size(0);
-  int smallest_engine = 0;
-  for (const auto i : cached_engine_batches_) {
-    if (i >= num_batch) {
-      smallest_engine = i;
-      break;
-    }
-  }
-  // TODO(sami): Need an LRU here
-  if (smallest_engine == 0) {
-    if (max_cached_engines_ > cached_engine_batches_.size()) {
-      smallest_engine = num_batch;
-      cached_engine_batches_.push_back(num_batch);
-      VLOG(1) << "Running with batch size " << num_batch;
-    } else {
-      string msg =
-          StrCat("Engine buffer is full. buffer limit=", max_cached_engines_,
-                 ", current entries=");
-      for (auto i : cached_engine_batches_) StrAppend(&msg, i, ",");
-      StrAppend(&msg, " requested batch=", num_batch);
-      LOG(WARNING) << msg;
-      return -1;
+bool TRTEngineOp::GetCompatibleCachedEngine(
+    const std::vector<TensorShape>& input_shapes,
+    std::vector<TensorShape>* matched_shape_out) {
+  const int batch_size = input_shapes[0].dim_size(0);
+  int smallest_batch_size = -1;
+  // Output shape will always be the same as the input but we will overwrite the
+  // batch size.
+  *matched_shape_out = input_shapes;
+  for (const int cached_batch_size : cached_engine_batch_sizes_) {
+    // Check if compatible: batch <= cached batch.
+    if (batch_size <= cached_batch_size) {
```

Comment:
Just curious: here we only compare the first dim a.k.a the batch size, do we plan to support different non-batch dimensions? Please add comment for that.